src/tool_operate.c

```diff
@@ -653,8 +653,12 @@ static CURLcode post_per_transfer(struct GlobalConfig *global,
       errorf(config->global, "curl: (%d) Failed writing body", result);
     }
     if(result && config->rm_partial) {
-      notef(global, "Removing output file: %s", outs->filename);
-      unlink(outs->filename);
+      struct_stat st;
+      stat(outs->filename, &st);
+      if(S_ISREG(st.st_mode)) {
```

Comment:
The stat error code should be checked or this could be working on uninitialized data.

Reply:
Indeed!